### ISSUE TITLE
Old errors still showing

### ISSUE THREAD

**Issue by robertmoura** [yoavbls/pretty-ts-errors#3]
Love the error message format but I noticed the old ones are still showing up. Was this intended or is there a way to remove them?

<img width="385" alt="Screenshot 2023-04-13 at 9 04 08 am" src="https://user-images.githubusercontent.com/17804984/231604904-87de80f2-9cf7-4573-a5f0-88ad1128a4fb.png">

If the environment might have something to do with it I am using Vue with Volar in TypeScript takeover mode. I saw johnsoncodehk has contributed recently which is great!

**Comment by Samathingamajig** [comment]
this is intended.
https://github.com/yoavbls/pretty-ts-errors/blob/d055b017030f3c24a2c70fc74cc785d3fa49ada5/README.md?plain=1#L38
> VSCode markdown blocks all styling options, so I had to find hacks to style the error messages. e.g., there isn't an inlined code block on VSCode markdown, so I used a code block inside a codicon icon, which is the only thing that can be inlined. That's why it can't be copied. but it isn't a problem because you can still hover on the error and copy things from the original error pane.

**Comment by seahindeniz** [comment]
Hmm. Well, if it's possible, having an option to disable them is much preferred than having a bunch of similar errors in a scrollable small container. For the copy-pasting issue, I believe the error translation feature and also the built-in `Problems` panel should be enough, since it shows the raw version of the error.

**Comment by robertmoura** [comment]
@seahindeniz I agree! Would be great to have an option to disable them. Personally I never copy these errors anyway.

**Comment by Samathingamajig** [comment]
at most, i copy the error number, in this case `ts(6196)` or just `6196` and type `ts` in the search bar

**Comment by yoavbls** [comment]
@robertmoura  Thank you for opening this issue 💁🏼‍♀️
As @Samathingamajig  said, it is intended, but I planned to add an option to choose whether you want to see TypeScript original errors after I'll succeed in making the types copyable.
Unfortunately, I failed to find a hack due to the strict limitations there:
https://github.com/microsoft/vscode/blob/b0d7acec38cfab4d82630cba64d581bf1cd1a305/src/vs/base/browser/markdownRenderer.ts#L372

So, I'll explore adding a setting that will hide TypeScript's original errors very soon.
Until then, if someone want to open a PR for that, you're more than welcome 🙂

**Comment by Samathingamajig** [comment]
Are extensions able to silence the errors of another extension?

**Comment by yoavbls** [comment]
@Samathingamajig Actually I didn't find any API of VSCode that allows that.
Does someone have an idea how we can remove the original TypeScript diagnostics?

Maybe we should address it with a [language service plugin](https://github.com/microsoft/TypeScript/wiki/Writing-a-Language-Service-Plugin) like this guy did [here](https://github.com/zardoy/vscode-disable-ts-errors/tree/main)?
@johnsoncodehk @Samathingamajig @tjx666 
WDYT?

**Comment by Samathingamajig** [comment]
> Maybe we should address it with a [language service plugin](https://github.com/microsoft/TypeScript/wiki/Writing-a-Language-Service-Plugin) like this guy did [here](https://github.com/zardoy/vscode-disable-ts-errors/tree/main)?
> @Samathingamajig WDYT?

Personally, I don't think it's worth it to try to silence the original error. Since formatting restricts the ability to copy/paste the error with the customize one, you get the benefit of being able to easily copy the raw error without going to the error panel. I would imagine silencing the error with a language service plugin would also remove the error from the error panel, as well as hiding it from your extension (unless your extension relies on a separate language server, which I doubt).

**Comment by nicooprat** [comment]
At least it would be cool to have the pretty error first in the tooltip 🤞

**Comment by yoavbls** [comment]
It was the first on the tooltip a while ago, I'm not sure why it changed.
@johnsoncodehk has some breakthroughs there by moving the error-providing mechanism to the language service plugin.
It'll allow us to control the order and hide the original errors.

We'll update about it soon 🙏🏼

**Comment by johnsoncodehk** [comment]
Just to clarify...don't expect it, the language server can cut the original error message, but this will also cause extensions like ErrorLens that rely on diagnostic messages to not work, so I don't think that's the way forward. Still need see VSCode how to improve.

In addition, the language server can only control the order of hover messages, but cannot make the hover messages displayed on top of the diagnosis.

Sorry for the false hopes.

**Comment by corysimmons** [comment]
Can someone just link to a VSCode repo Issue where we can 👍 some _"Let us tweak this"_ issue?

**Comment by estubmo** [comment]
> > Maybe we should address it with a [language service plugin](https://github.com/microsoft/TypeScript/wiki/Writing-a-Language-Service-Plugin) like this guy did [here](https://github.com/zardoy/vscode-disable-ts-errors/tree/main)?
> > @Samathingamajig WDYT?
> 
> Personally, I don't think it's worth it to try to silence the original error. Since formatting restricts the ability to copy/paste the error with the customize one, you get the benefit of being able to easily copy the raw error without going to the error panel. I would imagine silencing the error with a language service plugin would also remove the error from the error panel, as well as hiding it from your extension (unless your extension relies on a separate language server, which I doubt).

A good solution for that would be to hide the original error, but add a button for copying it all to the clipboard.

**Comment by Samathingamajig** [comment]
> A good solution for that would be to hide the original error, but add a button for copying it all to the clipboard.

read the last sentence

**Comment by wenfangdu** [comment]
I'm having this issue too, the code is still collapsed when using Volar takeover mode:
![image](https://user-images.githubusercontent.com/28700378/233266226-fab44a51-f566-4a78-8c99-46d08bdcb4ac.png)

**Comment by RoyiNamir** [comment]
For being fare , don't display just the beautified error in the print screen in the main screen of the extension. show the FULL output.

You're basically lying. people expect to see like what's in your marketplace , but actually it's not like that.

**Comment by Samathingamajig** [comment]
@RoyiNamir It could be made more prominent, however, it is mentioned on the main screen of the extension.

https://github.com/yoavbls/pretty-ts-errors/blob/d055b017030f3c24a2c70fc74cc785d3fa49ada5/README.md?plain=1#L38

> VSCode markdown blocks all styling options, so I had to find hacks to style the error messages. e.g., there isn't an inlined code block on VSCode markdown, so I used a code block inside a codicon icon, which is the only thing that can be inlined. That's why it can't be copied. ***but it isn't a problem because you can still hover on the error and copy things from the original error pane.***

(emphasis mine)

**Comment by daaanny90** [comment]
@RoyiNamir A bit rude but I was expecting also only the beautified error. That's why I found this issue.

**Comment by yoavbls** [comment]
This is the issue that will eventually allow us to modify the order of the errors + mute the original error:
https://github.com/microsoft/vscode/issues/54272

### Every vote or comment could help!

@robertmoura Can you add it to the content of the issue please? 🙏🏼

**Comment by blasco** [comment]
This would be great, voting 👍

**Comment by scottybrown** [comment]
If this is expected behaviour, perhaps the README.md could be updated to better describe it.
<img width="821" alt="Screenshot 2023-11-12 at 3 40 06 pm" src="https://github.com/yoavbls/pretty-ts-errors/assets/3635061/3a98d955-e423-4125-bf58-33308bef0441">
This extension doesn't give you "this" "instead of that". It gives you both, and the ugly "that" gets priority. The only way to find out it's meant to work that way is to browse the issues :)

**Comment by oscdot** [comment]
I can confirm that the original "ugly" typescript errors in fact do appear over the "pretty" ones for me as well, feels like they were on top before, so might be as of lately that pretty ones appear below, sadly. For long error messages, this quickly makes the extension near useless, as scrolling the tiny little problems window is far from easy.

I would agree that the README should be updated to more accurately show how the errors will be displayed.

Hope that VSCode will allow for a way to work around this. It's sad, since the extension is def one of my favourites 🙏🏽

**Comment by yoavbls** [comment]
Until things will be changed, you can use this hack here:
https://github.com/yoavbls/pretty-ts-errors/blob/main/docs/hide-original-errors.md

**Comment by fabulousgk** [comment]
Hmm, I am able to copy from the prettified text.  macOS.

**Comment by yoavbls** [comment]
> Hmm, I am able to copy from the prettified text. macOS.

I'm curious how. Are you able to copy just texts or also colored code blocks?

**Comment by fabulousgk** [comment]
> I'm curious how. Are you able to copy just texts or also colored code blocks?

Ahh, now I see the issue!  It does seem that at least one of the supporting issues has been upvoted enough, but even though that occurred does not seem like it is getting done, which is  shame.

**Comment by rauf322** [comment]
Your script for hiding element doesn't work for me in cursor maybe the code blocks are different

**Comment by rauf322** [comment]
for me on the cursor everything was even easier just add 
div[tabindex="0"] .hover-row .marker.hover-contents {
    display: none;
}
and the element will disappears and u copy and paste elements with no problems

**Comment by vasilescur** [comment]
How to use with Cursor:

Install this extension: https://open-vsx.org/extension/subframe7536/custom-ui-style

Add this to `settings.json`:

```json
"custom-ui-style.stylesheet": {
    ".monaco-hover-content .hover-row .marker.hover-contents": {
      "display": "none !important"
    },
    ".monaco-hover-content .fix-buttons-container": {
      "display": "none !important"
    },
    ".monaco-hover-content .hover-row.status-bar": {
      "display": "none !important"
    }
}
```

**Comment by bjornlll** [comment]
@vasilescur that didn't work for me. maybe the CSS class names have changed

<img width="416" height="324" alt="Image" src="https://github.com/user-attachments/assets/a88ee22c-32c9-4d19-90fd-3cf632946ec1" />
